### PR TITLE
git: Remove external calls to git.ExecSafe

### PR DESCRIPTION
**Pull request by ryanslade**
Remove all uses of git.ExecSafe from outside `vcs/git`.

To ensure it can't be used, we also unexport it. This change is part
of a move to isolating "raw" git commands to a single package.

Closes https://github.com/sourcegraph/sourcegraph/issues/28181

### CODE REVIEW

File: cmd/frontend/graphqlbackend/externallink/repository.go [hunk 1]
```diff
@@ -44,9 +43,8 @@ func FileOrDir(ctx context.Context, db database.DB, repo *types.Repo, rev, path
 	phabRepo, link, serviceType := linksForRepository(ctx, db, repo)
 	if phabRepo != nil {
 		// We need a branch name to construct the Phabricator URL.
-		branchName, _, _, err := git.ExecSafe(ctx, repo.Name, []string{"symbolic-ref", "--short", "HEAD"})
-		branchName = bytes.TrimSpace(branchName)
-		if err == nil && string(branchName) != "" {
+		branchName, _, err := git.GetDefaultBranchShort(ctx, repo.Name)
+		if err == nil && branchName != "" {
 			links = append(links, NewResolver(
 				fmt.Sprintf("%s/source/%s/browse/%s/%s;%s", strings.TrimSuffix(phabRepo.URL, "/"), phabRepo.Callsign, url.PathEscape(string(branchName)), path, rev),
```
Comment (indradhanush) [inline cmd/frontend/graphqlbackend/externallink/repository.go:49]: Fix unnecessary conversion caught in lint.

```suggestion
				fmt.Sprintf("%s/source/%s/browse/%s/%s;%s", strings.TrimSuffix(phabRepo.URL, "/"), phabRepo.Callsign, url.PathEscape(branchName), path, rev),
```

**Review by indradhanush — APPROVED** [review]
LGTM pending fix for error caught in linter.

File: cmd/frontend/graphqlbackend/externallink/repository.go (hunk 1 above)
Comment (ryanslade) [reply cmd/frontend/graphqlbackend/externallink/repository.go:49]: Thanks, I already pushed that while you were reviewing I think.

Comment (indradhanush) [reply cmd/frontend/graphqlbackend/externallink/repository.go:49]: Ah, I see. It doesn't show up here though. :thinking:

**Pull request merged**